### PR TITLE
fix: Python API event client wait_for_result_availability method

### PR DESCRIPTION
**Pull request by qdelamea-aneo**
- fix: Python API event client wait_for_result_availability method
- fix: Python event client

### CODE REVIEW

File: packages/python/src/armonik/client/events.py [hunk 1]
```diff
@@ -56,66 +57,84 @@ def get_events(
...
-            request.tasks_filters = cast(rawTaskFilters, task_filter.to_disjunction().to_message())
-        if result_filter:
-            request.results_filters = cast(
-                rawResultFilters, result_filter.to_disjunction().to_message()
-            )
+        request = EventSubscriptionRequest(
+            session_id=session_id,
+            returned_events=event_types,
+            tasks_filters=cast(rawTaskFilters, task_filter.to_disjunction().to_message())
+            if task_filter
```
Comment (dbrasseur-aneo) [inline packages/python/src/armonik/client/events.py:71]:
```suggestion
            if task_filter is not None
```
Filters to not implement "truthyness" as they can't be converted to booleans. Other instances of this are present in other parts of the file

File: packages/python/src/armonik/client/events.py [hunk 2]
```diff
@@ -56,66 +57,84 @@ def get_events(
...
-    def wait_for_result_availability(self, result_id: str, session_id: str) -> None:
+            for event_handler in event_handlers:
+                if event_handler(
+                    session_id,
+                    EventTypes.from_string(event_type),
+                    self._events_obj_mapping[event_type].from_raw_event(
+                        getattr(message, event_type)
+                    ),
+                ):
+                    break
```
Comment (dbrasseur-aneo) [inline packages/python/src/armonik/client/events.py:89]: This does not do what the docstring describes.
When an handler return True, it stops processing other handlers, but it doesn't stop listening as the "break" does not break the message loop.
you could use :  
```suggestion
            if all(not event_handler(
                session_id,
                EventTypes.from_string(event_type),
                self._events_obj_mapping[event_type].from_raw_event(
                    getattr(message, event_type)
                ),
            ) for event_handler in event_handlers):
                break
```

File: packages/python/src/armonik/client/events.py [hunk 3]
```diff
@@ -56,66 +57,84 @@ def get_events(
...
+
+    def wait_for_result_availability(self, result_ids: List[str], session_id: str) -> None:
```
Comment (dbrasseur-aneo) [inline packages/python/src/armonik/client/events.py:91]: I would keep the old one too with a
```suggestion
    def wait_for_result_availability(self, result_ids: Union[str, List[str]], session_id: str) -> None:
```
And wrap the singular one in a list
for Api compatibility

File: packages/python/src/armonik/client/events.py [hunk 4]
```diff
@@ -56,66 +57,84 @@ def get_events(
...
-            return
-        elif result.status == ResultStatus.ABORTED:
-            raise RuntimeError(f"Result {result.name} with ID {result_id} is aborted.")
-
-        self.get_events(
-            session_id,
-            [EventTypes.RESULT_STATUS_UPDATE],
-            [handler],
-            result_filter=(ResultFieldFilter.RESULT_ID == result_id),
+        results_not_found = deepcopy(result_ids)
```
Comment (dbrasseur-aneo) [inline packages/python/src/armonik/client/events.py:102]: A simple copy should be sufficient, as a string is immutable in Python

File: packages/python/src/armonik/client/events.py [hunk 5]
```diff
@@ -56,66 +57,84 @@ def get_events(
...
+
+        results_filter = ResultFieldFilter.RESULT_ID == result_ids[0]
```
Comment (dbrasseur-aneo) [inline packages/python/src/armonik/client/events.py:104]: Needs a check for empty list

**Review by dbrasseur-aneo — CHANGES_REQUESTED** [review]

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**